samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -39,12 +43,42 @@ static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 		if (request_value[1] == 0x03) {
 			/* Request for feature report */
 			if (request_value[0] == REPORT_ID_USER_CONFIG) {
-				static u8_t status;
-
-				status = atomic_get(&forward_status);
-				*len = sizeof(status);
-				*data = (u8_t *)(&status);
-
+				size_t length = *len;
+				u8_t *buffer = *data;
+
+				struct config_channel_frame frame;
+				int pos = config_channel_report_parse(buffer, length, &frame, true);
+				if (pos < 0) {
+					LOG_WRN("Could not parse report");
+					return pos;
+				}
+
+				bool is_fetch = (frame.status == FORWARD_STATUS_FETCH);
+				if (is_fetch) {
+					frame.status = atomic_get(&fetch_done) ? FORWARD_STATUS_SUCCESS : FORWARD_STATUS_PENDING;
+					pos = config_channel_report_fill(buffer, length,
+									 &frame, true);
+					if (pos < 0) {
+						LOG_WRN("Could not set report");
+						return pos;
+					}
+
+					if (atomic_get(&fetch_done)) {
```

Comment:
`atomic_get(&fetch_done)` may return a different value then the call in line 58 and lead to inconsistent message.
better to have a local copy of this (i.e. call `atomic_get` only once and operate on the result)

Reply:
done